tests/transforms/test_hamiltonian_expand.py

```diff
@@ -259,7 +260,287 @@ def test_hamiltonian_dif_tensorflow(self):
+        """Tests that the sum_expand transform returns the correct value
+        if we switch grouping off"""
+
+        tapes, fn = sum_expand(tape, group=False)
+        results = dev.batch_execute(tapes)
+        expval = fn(results)
+
+        assert all(qml.math.allclose(o, e) for o, e in zip(output, expval))
+
+        qs = QuantumScript(tape.operations, tape.measurements)
+        tapes, fn = sum_expand(qs, group=False)
+        results = dev.batch_execute(tapes)
+        expval = fn(results)
+
+        assert all(qml.math.allclose(o, e) for o, e in zip(output, expval))
+
+    def test_grouping(self):
+        """Test the grouping functionality"""
+        S = qml.op_sum(qml.PauliZ(0), qml.s_prod(2, qml.PauliX(1)), qml.s_prod(3, qml.PauliX(0)))
+
+        with QuantumTape() as tape:
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PauliX(wires=2)
+            qml.expval(S)
+
+        tapes, fn = sum_expand(tape, group=True)
+        assert len(tapes) == 2
+
+        qs = QuantumScript(tape.operations, tape.measurements)
+        tapes, fn = sum_expand(qs, group=True)
+        assert len(tapes) == 2
+
+    def test_number_of_tapes(self):
+        """Tests that the the correct number of tapes is produced"""
+
+        S = qml.op_sum(qml.PauliZ(0), qml.s_prod(2, qml.PauliX(1)), qml.s_prod(3, qml.PauliX(0)))
+
+        with QuantumTape() as tape:
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PauliX(wires=2)
+            qml.expval(S)
+
+        tapes, fn = sum_expand(tape, group=False)
+        assert len(tapes) == 3
+
+        tapes, fn = sum_expand(tape, group=True)
+        assert len(tapes) == 2
+
+    def test_number_of_qscripts(self):
+        """Tests the correct number of quantum scripts are produced."""
+
+        S = qml.op_sum(qml.PauliZ(0), qml.s_prod(2, qml.PauliX(1)), qml.s_prod(3, qml.PauliX(0)))
+        qs = QuantumScript(measurements=[qml.expval(S)])
+
+        tapes, fn = sum_expand(qs, group=False)
+        assert len(tapes) == 3
+
+        tapes, fn = sum_expand(qs, group=True)
+        assert len(tapes) == 2
+
+    def test_non_sum_tape(self):
+        """Test that the ``sum_expand`` function returns the input tape if it does not
+        contain a single measurement with the expectation value of a Sum."""
+
+        with QuantumTape() as tape:
+            qml.expval(qml.PauliZ(0))
+
+        tapes, fn = sum_expand(tape)
+
+        assert len(tapes) == 1
+        assert isinstance(list(tapes[0])[0].obs, qml.PauliZ)
+        # Old returntypes return a list for a single value:
+        # e.g. qml.expval(qml.PauliX(0)) = [1.23]
+        res = [1.23] if qml.active_return() else [[1.23]]
+        assert fn(res) == 1.23
+
+    def test_multiple_sum_tape(self):
+        """Test that the ``sum_expand`` function can expand tapes with multiple sum observables"""
+
+    @pytest.mark.autograd
+    def test_sum_dif_autograd(self, tol):
```

Comment:
Worth adding a jax test? I think that will be our interface of choice in the coming terms

Reply:
Added!